tests/system/action/motion/test_update_metadata.py

```diff
@@ -38,6 +38,45 @@ def test_update_correct(self) -> None:
         assert model.get("tag_ids") == []
         assert model.get("attachment_ids") == []
 
+    def test_update_metadata_workflow_id(self) -> None:
+        self.create_model("meeting/2538", {"name": "name_jkPIYjFz"})
+        self.create_model(
+            "motion/111", {"name": "name_srtgb123", "meeting_id": 2538, "state_id": 88}
+        )
+        self.create_model(
+            "motion_workflow/22", {"name": "name_workflow_22", "meeting_id": 2538}
+        )
+        self.create_model(
+            "motion_state/88",
+            {
+                "name": "name_blaglup",
+                "meeting_id": 2538,
+                "workflow_id": 22,
+                "motion_ids": [111],
+            },
+        )
+        self.create_model(
+            "motion_state/23",
+            {"name": "name_state_23", "meeting_id": 2538, "motion_ids": []},
+        )
+        self.create_model(
+            "motion_workflow/35",
+            {"name": "name_workflow_35", "first_state_id": 23, "meeting_id": 2538},
+        )
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "motion.update_metadata",
+                    "data": [{"id": 111, "workflow_id": 35}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        model = self.get_model("motion/111")
+        assert model.get("state_id") == 23
+        assert model.get("recommendation_id") is None
+
```

Comment:
Please also add a test where the workflows don't differ and recommendation_id is set and make sure that neither recommendation_id nor state_id is changed.

Reply:
Done. Added a test.